sdk/core/azure-core/src/main/java/com/azure/core/util/polling/SyncPoller.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.polling;
+import java.time.Duration;
+
+/**
+ * Type for doing synchronous polling of a long-running operation.
+ *
+ * @param <T> The type of poll response value.
+ * @param <U> The type of the final result of long-running operation.
+ */
+public interface SyncPoller<T, U> {
+    /**
+     * Poll once and return the poll response received.
+     *
+     * @return the poll response
+     */
+    PollResponse<T> poll();
```

Comment:
Does this force another poll to the server, or does it just return the last poll response? If it forces a poll, this may lead to people overwhelming services with poll requests.

Reply:
I believe this forces another poll - there's no backend thread running constantly to poll the latest status.

Reply:
If the user thinks it's a force REST call, they probably won't wrap this method in a busy loop. On the other hand, if the user thinks this method just checks the status of the last poll (while some thread is polling in the background), they probably won't check the status in a busy loop too. So I'm not sure if your concern maps to a real world scenario.

Reply:
User shouldn't use tight loop and must honor retry-after to handle any 429.